india_banking/public/js/payment_order.js

```diff
@@ -1,363 +1,512 @@
+    const existing_payment_requests = (frm.doc.references || []).map(
+      (reference) => reference.payment_request
+    );
+
+    // Use map_current_doc utility to fetch and map payment requests
+    erpnext.utils.map_current_doc({
+      method: "india_banking.overrides.payment_request.make_payment_order",
+      source_doctype: "Payment Request",
+      target: frm,
+      setters: {
+        party_type: "",
+        party: "",
+        grand_total: "",
+      },
+      get_query_filters: {
+        docstatus: 1,
+        status: ["=", "Initiated"],
+        bank: frm.doc.bank,
+        name: ["not in", existing_payment_requests],
+        company: frm.doc.company,
+      },
+    });
+  },
+
+  get_payments_from_payment_entry(frm) {
+    // Ensure references table is clean before processing
+    frm.trigger("remove_row_if_empty");
+
+    // Collect existing payment entries from the references table, if any
+    const existing_payment_entries = (frm.doc.references || []).map(
+      (reference) => reference.payment_entry
+    );
+
+    // Use map_current_doc utility to fetch and map payment entries
+    erpnext.utils.map_current_doc({
+      method: "india_banking.overrides.payment_entry.make_payment_order",
+      source_doctype: "Payment Entry",
+      target: frm,
+      setters: {
+        party: "",
+        paid_amount: "",
+      },
+      get_query_filters: {
+        docstatus: 1,
+        name: ["not in", existing_payment_entries],
+        source_doctype: ["!=", "Payment Request"],
+        payment_type: "Pay",
+        mode_of_payment: "Wire Transfer",
+        bank_account: frm.doc.company_bank_account,
+      },
+    });
+  },
+
+  get_payments_from_journal_entry(frm) {
+    erpnext.utils.map_current_doc({
+      method: "india_banking.overrides.journal_entry.make_payment_order",
+      source_doctype: "Journal Entry",
+      target: frm,
+      setters: [
+        {
+          fieldtype: "Link",
+          label: "Company",
+          fieldname: "company",
+          options: "Company",
+          default: frappe.defaults.get_user_default("company"),
+        },
+        {
+          fieldtype: "Select",
+          label: "Entry Type",
+          fieldname: "voucher_type",
+          options: "Bank Entry",
+          hidden: true,
+        },
+        {
+          fieldtype: "Currency",
+          label: "Amount",
+          fieldname: "total",
+          hidden: true,
+        },
+      ],
+      get_query: function () {
+        // Extract unique reference names from the references table
+        const existing_journal_entries = [
+          ...new Set(
+            (frm.doc.references || []).map(
+              (reference) => reference.reference_name
+            )
+          ),
+        ];
+        return {
+          query: "india_banking.overrides.journal_entry.get_bank_entry",
+          filters: {
+            docs: existing_journal_entries,
+            company_account: frm.doc.account,
```

Comment:
posting_date lesser than current date, currency being INR, only those lines that are involving the company bank account of the payment order, are these handled?